assets/src/edit-story/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -166,9 +173,26 @@ function useMediaUploadQueue() {
 
             try {
               const attachment = await uploadFile(file, additionalData);
+              const { poster, posterId } = await processPosterData(
+                attachment.id,
+                fileName,
+                posterFile
+              );
+
+              // The newly uploaded file won't have a poster yet.
+              // However, we'll likely still have one on file.
+              // Add it back so we're never without one.
+              // The final poster will be uploaded later by uploadVideoPoster().
+              const newResource = getResourceFromAttachment(attachment);
+              const newResourceWithPoster = {
+                ...newResource,
+                poster: poster || newResource.poster || resource.poster,
+                posterId,
+              };
```

Comment:
Adding this here, as is makes the logic as the transcoded version. Not required, but a nice improvement.